src/channel/ably/utils.ts

```diff
@@ -0,0 +1,174 @@
+export const isNullOrUndefined = (obj) => obj == null || obj === undefined;
+export const isEmptyString = (stringToCheck, ignoreSpaces = true) => (ignoreSpaces ? stringToCheck.trim() : stringToCheck) === '';
+export const isNullOrUndefinedOrEmpty = (obj) => obj == null || obj === undefined || isEmptyString(obj);
+
+/**
+ * @throws Exception if parsing error
+ */
+export const parseJwt = (jwtToken: string): { header: any, payload: any } => {
+    // Get Token Header
+    const base64HeaderUrl = jwtToken.split('.')[0];
+    const base64Header = base64HeaderUrl.replace('-', '+').replace('_', '/');
+    const header = JSON.parse(toText(base64Header));
+    // Get Token payload
+    const base64Url = jwtToken.split('.')[1];
+    const base64 = base64Url.replace('-', '+').replace('_', '/');
+    const payload = JSON.parse(toText(base64));
+    return { header, payload };
+}
+
+export const toTokenDetails = (jwtToken: string) => {
+    const { payload } = parseJwt(jwtToken);
+    return {
+        // capability: parsedJwt['x-ably-capability'], // RSA4f - tokenDetails size should't exceed 128kb
+        clientId: payload['x-ably-clientId'],
+        expires: payload.exp * 1000, // Convert Seconds to ms
+        issued: payload.iat * 1000,
+        token: jwtToken
+    };
+}
+
+const isBrowser = typeof window === 'object';
+const isNativescript = typeof global === 'object' && global.isNativescript;
+
+const toBase64 = (text: string) => {
+    if (isBrowser) {
+        return btoa(text);
+    }
+    return Buffer.from(text, 'binary').toString('base64');
+};
+
+const toText = (base64: string) => {
+    if (isBrowser) {
+        return atob(base64);
+    }
+    return Buffer.from(base64, 'base64').toString('binary');
+};
+
+function createXHR() {
+    var result = new XMLHttpRequest();
+    if ('withCredentials' in result) return result;// @ts-ignore
+    if (typeof XDomainRequest !== 'undefined') { // @ts-ignore
+        var xdr = new XDomainRequest(); /* Use IE-specific "CORS" code with XDR */
+        xdr.isXDR = true;
+        return xdr;
+    }
+    return null;
+}
+
+function schemeMatchesCurrent(scheme) {
+    return scheme === window.location.protocol.slice(0, -1);
+}
+
+export function httpReqFunction() {
```

Comment:
I see this was lifted out of the ably-js test suite, but it was designed to support NodeJS and NativeScript - we should remove those bits since project this will only run in web browsers

Reply:
Yeah, actually it's needed to run tests! If there is a way to inject `httpClient` externally via tests, that would be great 👍 . I will check if it's possible.

Reply:
I have updated this with a fix!